SpatialGDK/Source/SpatialGDK/Private/Interop/CrossServerRPCSender.cpp

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/CrossServerRPCSender.h"
+
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Utils/SpatialMetrics.h"
+
+using namespace SpatialGDK;
+
+CrossServerRPCSender::CrossServerRPCSender(ViewCoordinator& InCoordinator, USpatialMetrics* InSpatialMetrics)
+	: Coordinator(InCoordinator)
+	, SpatialMetrics(InSpatialMetrics)
+{
+}
+
+void CrossServerRPCSender::SendCommand(const FUnrealObjectRef& InTargetObjectRef, UObject* TargetObject, UFunction* Function,
+									   RPCPayload&& InPayload, FRPCInfo Info, const TOptional<Trace_SpanId>& SpanId) const
+{
+	if (Function == nullptr || TargetObject == nullptr || InTargetObjectRef.Entity == SpatialConstants::INVALID_ENTITY_ID
+		|| Info.Type != ERPCType::CrossServer)
+	{
+		return;
+	}
+
+	const auto SchemaType = Schema_CreateCommandRequest();
+	Schema_Object* RequestObject = Schema_GetCommandRequestObject(SchemaType);
+	RPCPayload::WriteToSchemaObject(RequestObject, InTargetObjectRef.Offset, Info.Index, Function->GetUniqueID(),
```

Comment:
one thing I'm not sure about: is getting the unique id of the function enough? what if the same rpc gets triggered twice on purpose?

Reply:
I think you should use a random number.